### ISSUE TITLE
✨ Enhancement: DocsHelp component option to hide Edit This on GitHub

### ISSUE THREAD

**Issue by DarhkVoyd** [json-schema-org/website#1099]
### Is your feature request related to a problem? Please describe

The pages using DocsHelp component, have a link "Edit this page on GitHub". In case of `/tools` it goes to https://github.com/json-schema-org/website/blob/main/pages/%2Ftools/index.page.tsx, which seems to just be the layout of the page, not the content itself. As discussed most of the time people would want to add or edit entries for their own tool.

### Describe the solution you'd like

We can add an option to feedback component (DocsHelp) to disable/hide the `Edit this page on GitHub` or an option to pass a custom link, in this case the tooling data.

### Are you working on this?

No

**Comment by DarhkVoyd** [comment]
@techmannih Sure, go ahead.

**Comment by arpitkuriyal** [comment]
@DarhkVoyd  can you please assign this to me?

**Comment by DarhkVoyd** [comment]
@arpitkuriyal Yes, please go ahead. @benjagm will assign you the issue soon.

**Comment by arpitkuriyal** [comment]
@DarhkVoyd  i have make a prop to hide the "edit this page in the github" . Can you tell me in which pages i should hide it and on which pages i should  show it?
In part where the prop is passed  with false value the page will  look like this
<img width="1468" alt="Screenshot 2024-11-13 at 6 27 15 PM" src="https://github.com/user-attachments/assets/82b5f3ff-9d24-4d9a-9bfa-b2edc63049aa">
is it fine?

**Comment by DarhkVoyd** [comment]
@arpitkuriyal Good work on your progress. Let's not hide the entire section instead just hiding the button would be better. The default behavior for the `DocsHelp` component should be to show the button, that way we will only have to manually handle cases when we want to hide it. For now, we only want to hide the button on the `/tools` page.

**Comment by arpitkuriyal** [comment]
@DarhkVoyd Okay okay I got it

**Comment by arpitkuriyal** [comment]
@gregsdennis Yes, there's a prop that allows specifying a custom redirect link in the DocsHelp component. However, based on the issue's requirements, it doesn't need to be included on the tooling page . please let me know if you have more suggestions.

**Comment by gregsdennis** [comment]
I'm suggesting the issue's requirements might be the wrong thing to do. This is merely a proposal. The issue is for discussion. Issues aren't just to-dos.

@benjagm, please weigh in here.

**Comment by gregsdennis** [comment]
@DarhkVoyd the opening comment says "as discussed" but doesn't link to where anything was discussed.

**Comment by arpitkuriyal** [comment]
@gregsdennis  sorry but i am just a beginner learning things i see this is a good first issue so i picked it. i didn't see that it is discuss or not  or just a proposal to be discussed.

**Comment by benjagm** [comment]
> @DarhkVoyd the opening comment says "as discussed" but doesn't link to where anything was discussed.

Discussed [here](https://json-schema.slack.com/archives/C8C4UBXDF/p1731094162416679?thread_ts=1731094162.416679&cid=C8C4UBXDF) in slack.

**Comment by DarhkVoyd** [comment]
@gregsdennis The current behavior of the “Edit this page on GitHub” button is to navigate to the file that contains the page’s content. This is specified by a prop, which accepts the file type to construct the GitHub path. For instance, “_index” (referring to “_index.md”) or by default, it points to “index.page.tsx” for that path. However, for the “/tools” page, all the content is from a data file rather than a markdown file, so by default, it points to “index.page.tsx,” which primarily consists of the layout and logic. Currently, there’s no logic in place to use a custom link.

**Comment by gregsdennis** [comment]
Thanks, Benja.

@DarhkVoyd I understood the problem. What I was missing was the context that provided the solution.  Just looking at this issue, it seems like a decision was just made without discussion.  Let's be sure to provide as much context as possible in the future, okay?

**Comment by manya1632** [comment]
Hi @benjagm !
I went through the discussion and understood that the main concern is that on pages like /tools, the “Edit this page on GitHub” link points to the layout file instead of the actual data source users would want to edit.

If this issue is still open and needs follow-up (for example, supporting a custom edit link for data-driven pages or refining the existing behavior), I’d be happy to work on it. Please assign it to me.
